src/brewHandler.h

```diff
@@ -72,89 +69,75 @@ boolean brewPIDDisabled = false;                    // is PID disabled for delay
-                // set new button state 
-            }
-            else if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
-                // whatever the reading is at, it's been there for longer than the debounce
-                // delay, so take it as the actual current state:
-                
-                if (brewSwitchTrigger != reading) {
-                    brewSwitchTrigger = reading;
-                }
-            }
-            lastStateBrewTrigger = reading;
+    uint8_t brewSwitchReading = brewSwitch->isPressed();
 
+    if (BREWSWITCH_TYPE == Switch::TOGGLE) {
+        currStateBrewSwitch = brewSwitchReading;
+    }
+    else if (BREWSWITCH_TYPE == Switch::MOMENTARY) {
+        if (currBrewSwitchStateMomentary != brewSwitchReading) {
+            currBrewSwitchStateMomentary = brewSwitchReading;
+        }
+        
         // Convert trigger signal to brew switch state
-        switch (brewSwitchTriggerCase) {
+        switch (brewSwitchCase) {
             case 10:
-                if (brewSwitchTrigger == HIGH && machineState != kWaterEmpty ) {
-                    brewSwitchTriggerMillis = millis();
-                    brewSwitchTriggerCase = 20;
-                    debugPrintln("brewSwitchTriggerCase 10: HIGH");
+                if (currBrewSwitchStateMomentary == HIGH && machineState != kWaterEmpty ) {
+                    brewSwitchCase = 20;
+                    LOG(DEBUG, "brewSwitchCase 10: HIGH");
                 }
                 break;
 
             case 20:
                 // Only one push, brew
-                if (brewSwitchTrigger == LOW) {
+                if (currBrewSwitchStateMomentary == LOW) {
                     // Brew trigger
-                    brewSwitch = HIGH;
-                    brewSwitchTriggerCase = 30;
-                    debugPrintln("brewSwitchTriggerCase 20: Brew Trigger HIGH");
+                    currStateBrewSwitch = HIGH;
+                    brewSwitchCase = 30;
+                    LOG(DEBUG, "brewSwitchCase 20: Brew Trigger HIGH");
                 }
 
-                // Button more than brewTriggerLongPress pushed
-                if (brewSwitchTrigger == HIGH && (brewSwitchTriggerMillis + brewTriggerLongPress  <= millis()) && machineState != kWaterEmpty) {
-                    debugPrintln("brewSwitchTriggerCase 20: Manual Trigger - flushing");
-                    brewSwitchTriggerCase = 31;
-                    digitalWrite(PIN_VALVE, relayOn);
-                    digitalWrite(PIN_PUMP, relayOn);
+                // Button more than brewSwitchMomentaryLongPress pushed
+                if (currBrewSwitchStateMomentary == HIGH && brewSwitch->longPressDetected() && machineState != kWaterEmpty) {
+                    brewSwitchCase = 31;
+                    valveRelay.on();
+                    pumpRelay.on();
                     startingTime = millis();
+                    LOG(DEBUG, "brewSwitchCase 20: Manual Trigger - flushing");
```

Comment:
this should be log level INFO, like all brewing states ( Prefinfusion, pre pause, brew
maybe we should move the action (pumpe and vavle) into the brewstatemashine and not trigger this out of the brewswitch state mashine?

Reply:
I had them at `DEBUG` because printing out enum values of a switch case felt rather technical. Can we leave this for post-merging to decide or improve?

Reply:
I gonna put that on the roadmap, but I would like to get this output back as soon as possible. Without there is no log feedback for the long press brewswitch and flushing